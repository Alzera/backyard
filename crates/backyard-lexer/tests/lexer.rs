use backyard_lexer::{ error::LexError, lex, lex_eval };

#[test]
fn magic() {
  let tokens = lex_eval("__DIR__").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn type_test() {
  let tokens = lex_eval("int").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn id() {
  let tokens = lex_eval("sample_id").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_equal() {
  let tokens = lex_eval("$a =& $b = $c === $d == (fn() => $d)();").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_ambersand() {
  let tokens = lex_eval("$a &= $b & $c && $d;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_hash() {
  let tokens = lex_eval("#[attr]\n$a; # comment").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_question_mark() {
  let tokens = lex("<?php $a ??= ($b ?? $c) ? ($d ?: $e) : $f?->g; ?>>>>").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_percent() {
  let tokens = lex_eval("$a %= $b % $c;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_caret() {
  let tokens = lex_eval("$a ^= $b ^ $c;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_star() {
  let tokens = lex_eval("$a **= $b *= $c * $d ** $e;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_slash() {
  let tokens = lex_eval("$a = $b / $c;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_dot() {
  let tokens = lex_eval("$a = $b . $c;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_pipe() {
  let tokens = lex_eval("$a |= $b | $c || $d;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_minus() {
  let tokens = lex_eval("$a -= $b->c - $d-- - --$e;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_gt() {
  let tokens = lex_eval("$a >>= $b >= $c >> $d > $e;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_lt() {
  let tokens = lex_eval("$a <<= $b <= $c << $d < $e <> $f <=> $g;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_colon() {
  let tokens = lex_eval("$a = $b::c ? $d : $e;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_exclamation() {
  let tokens = lex_eval("$a = $b !== $c != !!$d;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_plus() {
  let tokens = lex_eval("$a += $b++ + ++$c;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_square_bracket() {
  let tokens = lex_eval("$a[1, 2, 3];").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_backslash() {
  let tokens = lex_eval("\\A\\B;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_tilde() {
  let tokens = lex_eval("$a = ~false;").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn test_at() {
  let tokens = lex_eval("$a = @$this();").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn unrecognized() {
  let tokens = lex_eval("class � {\n}");
  assert_eq!(tokens, Err(LexError::Unrecognized { token: "�".to_string(), line: 1, column: 7 }));
}
