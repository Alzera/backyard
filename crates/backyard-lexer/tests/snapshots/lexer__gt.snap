---
source: crates/backyard-lexer/tests/lexer.rs
expression: tokens
snapshot_kind: text
---
- token_type: Variable
  value:
    - 97
  line: 1
  column: 0
  offset: 0
- token_type: BitwiseShiftRightAssignment
  value:
    - 62
    - 62
    - 61
  line: 1
  column: 3
  offset: 3
- token_type: Variable
  value:
    - 98
  line: 1
  column: 7
  offset: 7
- token_type: IsGreaterOrEqual
  value:
    - 62
    - 61
  line: 1
  column: 10
  offset: 10
- token_type: Variable
  value:
    - 99
  line: 1
  column: 13
  offset: 13
- token_type: BitwiseShiftRight
  value:
    - 62
    - 62
  line: 1
  column: 16
  offset: 16
- token_type: Variable
  value:
    - 100
  line: 1
  column: 19
  offset: 19
- token_type: IsGreater
  value:
    - 62
  line: 1
  column: 22
  offset: 22
- token_type: Variable
  value:
    - 101
  line: 1
  column: 24
  offset: 24
- token_type: Semicolon
  value:
    - 59
  line: 1
  column: 26
  offset: 26
